src/fsharp/FSharp.Build/Microsoft.FSharp.Targets

```diff
@@ -400,6 +400,63 @@ this file.
 
     </Target>
 
+    <!--
+
+    The old reference assemblies are no longer installed to `Program Files (x86)\Reference Assemblies\Microsoft\FSharp`.
+
+    To avoid breaking legacy projects the hint path for FSharp.Core will be rewritten if and only if the existing hint
+    path is unchanged from the original legacy templates and if a new reference assembly can be found at the new location.
+
+    -->
+    <Target Name="RedirectFSharpCoreReferenceToNewRedistributableLocation" BeforeTargets="ResolveAssemblyReferences">
+      <PropertyGroup>
+        <_OldRootSdkLocation>$(MSBuildProgramFiles32)\Reference Assemblies\Microsoft\FSharp</_OldRootSdkLocation>
+        <_NewRootSdkLocation>$(VsInstallRoot)\MSBuild\15.0\Bin\FSharpSdk</_NewRootSdkLocation>
+
+        <_CoreRelativeSuffix>.NETCore\$(TargetFSharpCoreVersion)\FSharp.Core.dll</_CoreRelativeSuffix>
+        <_FrameworkRelativeSuffix>.NETFramework\v4.0\$(TargetFSharpCoreVersion)\FSharp.Core.dll</_FrameworkRelativeSuffix>
+        <_PortableRelativeSuffix>.NETPortable\$(TargetFSharpCoreVersion)\FSharp.Core.dll</_PortableRelativeSuffix>
+
+        <!-- .NETCore -->
+        <_OldCoreSdkPath>$(_OldRootSdkLocation)\$(_CoreRelativeSuffix)</_OldCoreSdkPath>
+        <_NewCoreSdkPath>$(_NewRootSdkLocation)\$(_CoreRelativeSuffix)</_NewCoreSdkPath>
+
+        <!-- .NETFramework\v4.0 -->
+        <_OldFrameworkSdkPath>$(_OldRootSdkLocation)\$(_FrameworkRelativeSuffix)</_OldFrameworkSdkPath>
+        <_NewFrameworkSdkPath>$(_NewRootSdkLocation)\$(_FrameworkRelativeSuffix)</_NewFrameworkSdkPath>
+
+        <!-- .NETPortable -->
+        <_OldPortableSdkPath>$(_OldRootSdkLocation)\$(_PortableRelativeSuffix)</_OldPortableSdkPath>
+        <_NewPortableSdkPath>$(_NewRootSdkLocation)\$(_PortableRelativeSuffix)</_NewPortableSdkPath>
+      </PropertyGroup>
+
+      <ItemGroup>
+        <!-- Update references to `.NETCore\*\FSharp.Core.dll`. -->
+        <Reference Update="%(Reference.Identity)"
+                   Condition="'%(Reference.Identity)' == 'FSharp.Core' and
```

Comment:
Does it matter if this misses references that were referenced by full assembly identity?

Reply:
These updates are only to catch project files that never changed the default values, and the F# templates (to my knowledge) have never used the full assembly identity.  If a developer modifies this value then they're on their own for locating FSharp.Core.